js/dragonex.js

```diff
@@ -0,0 +1,912 @@
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'active': true,
+                'precision': precision,
+                'taker': taker,
+                'limits': {
+                    'amount': {
+                        'min': market[4],
+                        'max': undefined,
+                    },
+                    'price': {
+                        'min': market[2],
+                        'max': undefined,
+                    },
+                    'cost': {
+                        'min': parseFloat (market[4]) * parseFloat (market[2]),
+                        'max': undefined,
+                    },
+                },
+            });
+        }
+        return result;
+    }
+
+    calculateFee (symbol, type, side, amount, price, takerOrMaker = 'taker', params = {}) {
+        const market = this.markets[symbol];
+        let key = 'quote';
+        const rate = market[takerOrMaker];
+        let cost = amount * rate;
+        let precision = market['precision']['price'];
+        if (side === 'sell') {
+            cost *= price;
+        } else {
+            key = 'base';
+            precision = market['precision']['amount'];
+        }
+        cost = this.decimalToPrecision (cost, ROUND, precision, this.precisionMode);
+        return {
+            'type': takerOrMaker,
+            'currency': market[key],
+            'rate': rate,
+            'cost': parseFloat (cost),
+        };
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        const response = await this.privatePostUserOwn (params);
+        const result = { 'info': response };
+        const balances = this.safeValue (response, 'data', []);
+        for (let i = 0; i < balances.length; i++) {
+            const balance = balances[i];
+            const currencyId = balance['coin_id'];
+            const code = this.safeCurrencyCode (currencyId);
+            const account = this.account ();
+            account['free'] = this.safeFloat (balance, 'volume') - this.safeFloat (balance, 'frozen');
+            account['used'] = this.safeFloat (balance, 'frozen');
+            account['total'] = this.safeFloat (balance, 'volume');
+            result[code] = account;
+        }
+        return this.parseBalance (result);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol_id': market['symbol_id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit; // default = maximum = 100
+        }
+        const response = await this.v1GetMarketDepth (this.extend (request, params));
+        const orderBook = this.parseOrderBook (response);
+        const buysList = [];
+        const sellsList = [];
+        for (let i = 0; i < this.safeValue (response, 'data', {})['buys'].length; i++) {
```

Comment:
Hi @caoshitong369 ! Can you please rewrite this bit to use the base `parseOrderBook` method? See how `fetchOrderBook` is implemented in most of the other exchanges. Thx!